From jasonuhl@sgi.com Fri Dec 17 20:03:46 2004
Date: Fri, 17 Dec 2004 20:03:47 -0800
From: Jason Uhlenkott <jasonuhl@sgi.com>
To: lkcd-devel@lists.sourceforge.net
Subject: [PATCH] Don't eliminate "unused" types from kerntypes.o
Message-ID: <20041218040346.GG192201@dragonfly.engr.sgi.com>
Mime-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
User-Agent: Mutt/1.4.2i
Status: RO
Content-Length: 755
Lines: 19

Don't eliminate "unused" debug types from init/kerntypes.o.  Without
this, we lack a bunch of important structure definitions, including
__dump_header and __dump_header_asm (which lcrash can't start at all
without).

-- 
Index: linux/init/Makefile
===================================================================
--- linux.orig/init/Makefile	2004-12-13 14:52:13.167673116 -0800
+++ linux/init/Makefile	2004-12-17 18:42:37.890562898 -0800
@@ -12,7 +12,7 @@
 extra-$(CONFIG_KERNTYPES)	+= kerntypes.o
 #For IA64, compile kerntypes in dwarf-2 format.
 ifeq ($(CONFIG_IA64),y)
-CFLAGS_kerntypes.o             := -gdwarf-2
+CFLAGS_kerntypes.o             := -gdwarf-2 -fno-eliminate-unused-debug-types
 else
 CFLAGS_kerntypes.o             := -gstabs
 endif

